Digitisers/SimHitMerge/src/SimHitMergeAlg.cpp

```diff
@@ -81,13 +85,43 @@ StatusCode SimHitMergeAlg::execute()
               }
               else id_vconb_map[id] = tmp_vconb;
                   
+              if(sanity_check){
+                  if ( test_id_hits_map.find(id) != test_id_hits_map.end()) test_id_hits_map[id].push_back(Simhit);
+                  else{
+                      std::vector<edm4hep::SimCalorimeterHit> tmp_vec ;
+                      tmp_vec.push_back(Simhit);
+                      test_id_hits_map[id] = tmp_vec;
+                  }
+              }
+
           }
-        
+
+          if(sanity_check){
+              for(std::map<unsigned long long, std::vector<edm4hep::SimCalorimeterHit> >::iterator iter = test_id_hits_map.begin(); iter != test_id_hits_map.end(); iter++){
+                  for(unsigned int i=0; i< iter->second.size(); i++){
+                     float pos1_x = iter->second.at(i).getPosition()[0];
+                     float pos1_y = iter->second.at(i).getPosition()[1];
+                     float pos1_z = iter->second.at(i).getPosition()[2];
+                     for(unsigned int j=i+1; j< iter->second.size(); j++){
+                         float pos2_x = iter->second.at(j).getPosition()[0];
+                         float pos2_y = iter->second.at(j).getPosition()[1];
+                         float pos2_z = iter->second.at(j).getPosition()[2];
+                         float dis = sqrt( (pos1_x-pos2_x)*(pos1_x-pos2_x) + (pos1_y-pos2_y)*(pos1_y-pos2_y) + (pos1_z-pos2_z)*(pos1_z-pos2_z) );
+                         if( dis > sqrt(10*10 + 10*10 + 1.2*1.2) )
```

Comment:
Please add { } around this code block. And also think to use variable instead of such magic numbers.

Reply:
what do you mean add {} ? I can change magic number to variables.

Reply:
You have `if` statement and another new line. It is better to write:
```cpp
if( dis > sqrt(10*10 + 10*10 + 1.2*1.2) ) {

}
```

Reply:
I see.